odoo/models.py

```diff
@@ -3659,23 +3660,24 @@ def _check_company(self, fnames=None):
                     regular_fields.append(name)
                 else:
                     property_fields.append(name)
+            elif field.name == 'company_id':
+                company_changed = True
 
-        if not (regular_fields or property_fields):
+        if not (regular_fields or property_fields or company_changed):
             return
 
         inconsistencies = []
+        IrRule = self.env['ir.rule'].sudo()
```

Comment:
why the sudo()?

Reply:
Because you need it to be able to access the rules